### PR TITLE
Feature copy datasets 

### PR DESCRIPTION
**Pull request by KutluOzel-b**
Please see this comment for context: https://github.com/zowe/vscode-extension-for-zowe/pull/2051#issuecomment-1431999564

---

## Proposed changes

add feature copy datasets ( pds, sequential, members across pds) with multi select capabilities.

## Release Notes

https://github.com/zowe/vscode-extension-for-zowe/issues/1550

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

---

Please see this comment for context: https://github.com/zowe/vscode-extension-for-zowe/pull/2051#issuecomment-1431999564

### CODE REVIEW

**Review by traeok — CHANGES_REQUESTED** [review]
Thanks for working on this feature @KutluOzel-b ! Just a couple things I've found while testing:

- Since the "Copy Data Set" option brings up a dialog for the new dataset(s) names and doesn't require the user to initiate the paste operation, should the feature be renamed to something like "Duplicate data set(s)" ? Also, would it be possible to add a title to the input box to describe its intent, e.g. "New name for copy of 'DATASET1'" ? This would help provide some clarity to the end-user when the input box appears.
- When right-clicking on data set members (PDS), the "Copy Member" option now says "Copy Data Set".
- Trying to copy one PS data set, and providing the name of an existing PS data set as the new name, results in a z/OSMF REST API error: 
image

In this situation we could probably disregard z/OSMF API errors, or just reject the operation before making the API request if the new name already matches one of the existing datasets.

**Comment by KutluOzel-b** [comment]
> Thanks for working on this feature @KutluOzel-b ! Just a couple things I've found while testing:
> 
> * Since the "Copy Data Set" option brings up a dialog for the new dataset(s) names and doesn't require the user to initiate the paste operation, should the feature be renamed to something like "Duplicate data set(s)" ? Also, would it be possible to add a title to the input box to describe its intent, e.g. "New name for copy of 'DATASET1'" ? This would help provide some clarity to the end-user when the input box appears.
> * When right-clicking on data set members (PDS), the "Copy Member" option now says "Copy Data Set".
> * Trying to copy one PS data set, and providing the name of an existing PS data set as the new name, results in a z/OSMF REST API error:
>   image
> 
> In this situation we could probably disregard z/OSMF API errors, or just reject the operation before making the API request if the new name already matches one of the existing datasets.

Hey @traeok , Thanks for review . 

I had a purpose while huddling copy features together which was make it more simple and let user control over the operation he/she does. I'm thinking having "copy member", "copy dataset", "duplicate datasets", "copy sequential datasets" is a not a good way of achieving good ux. 
At this point it is maybe good to have more feedbacks from the community/team to continue.

Regarding the attempt of copying already existing a datasets ; I'm not sure there is an api for checking if a dataset with selected name is existing or not and dismissing api errors might be confusing considering operation is done or not.

**Comment by traeok** [comment]
> I had a purpose while huddling copy features together which was make it more simple and let user control over the operation he/she does. I'm thinking having "copy member", "copy dataset", "duplicate datasets", "copy sequential datasets" is a not a good way of achieving good ux. At this point it is maybe good to have more feedbacks from the community/team to continue.

I agree that having too many different "Copy" operations would be redundant and confusing for the user - we could always stick with "Copy" here as the user is intentionally right-clicking on a dataset node anyway.

> Regarding the attempt of copying already existing a datasets ; I'm not sure there is an api for checking if a dataset with selected name is existing or not and dismissing api errors might be confusing considering operation is done or not.

In hindsight it is probably more complicated to ensure there isn't an overlap in existing/new data set names, so we could always leave that up to the user.

Going to do some quick tests to make sure it is working as intended and then I will update the review 👍

**Review by traeok — APPROVED** [review]

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
Thanks @KutluOzel-b this is working good.  I do have a couple things I wanted to mention.
were we changing the `Copy Dataset` to just `Copy`?  I remember the discussion and see paste was changed to just `Paste`.  Also, I have to click on the data set (closing & reopening) to see the newly added files.  Is there a way we can have this automatically refresh after paste is finished?

**Comment by KutluOzel-b** [comment]
> Thanks @KutluOzel-b this is working good. I do have a couple things I wanted to mention. were we changing the `Copy Dataset` to just `Copy`? I remember the discussion and see paste was changed to just `Paste`. Also, I have to click on the data set (closing & reopening) to see the newly added files. Is there a way we can have this automatically refresh after paste is finished?

Hey @JillieBeanSim, thanks for reviewing this. I changed copy datasets to just copy. Sorry that was my miss. 

Regarding refresh of ds contents ; I preserve the collapse state of partial datasets if the state is open then it could be seen new files are presenting. If closed yes user has to change the state by him/herself. Do you think we should open the expandable list in both cases ?

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (traeok) [inline packages/zowe-explorer/src/dataset/actions.ts:1506]: Instead of defining it and then returning this we can probably just `return ZoweExplorerApiRegister.getMvsApi...`

**Review by traeok — APPROVED** [review]
LGTM! Just left a comment about one of the return values

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (KutluOzel-b) [reply packages/zowe-explorer/src/dataset/actions.ts:1506]: Thanks @traeok that was for debugging purposes my miss.

**Comment by KutluOzel-b** [comment]
Hey @JillieBeanSim, Thanks for the feedback of refreshing ds after paste operation completes. 

https://user-images.githubusercontent.com/104970275/211808233-9c3fac74-bdbc-4ebb-b4b4-099c3d32316d.mov

Do you think should it behaves in above ss ? If not so, could you please provide an ss so I can take a look ? 

Thanks in advance

**Comment by JillieBeanSim** [comment]
@KutluOzel-b I think the behavior in SS is perfect.  Are those changes in?

**Review by JillieBeanSim — APPROVED** [review]

**Comment by JillieBeanSim** [comment]
Looks like the changes are in for the ds refreshing after paste.  Thanks @KutluOzel-b for the enhancement and for addressing all feedback

File: packages/zowe-explorer/package.json
Comment (t1m0thyj) [inline packages/zowe-explorer/package.json:112]: Should these commands be named consistently if they offer equivalent copy/paste behavior?
Either "copyDataSet" and "pasteDataSet", or "copyMember" and "pasteMember"

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:1528]: For the case where the destination DSN is identical to the original one (`lbl === text`), should we show a different error message?
Instead of "Enter valid dataset name", I think a more specific message like "Dataset name already exists" may be helpful.

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:1586]: Same question as above regarding having a more specific error message when "Dataset name already exists".

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (KutluOzel-b) [reply packages/zowe-explorer/src/dataset/actions.ts:1528]: Thanks @t1m0thyj  for reviewing this. We need to validate new Input too and I think changing the text might be confusing. I hesitated to add "Dataset name already exists" because of this. Otherwise user would expect something like "datasets cannot start with : " or "maximum 8 chars allowed" .. and so on I think user should extract the logic what is missing/wrong the operation I'm doing at the moment if it this not extremely hidden or irrevelant to current operation.

File: packages/zowe-explorer/package.json
Comment (KutluOzel-b) [reply packages/zowe-explorer/package.json]: Adressed in : a19a8de6821b088c5790d4b27484f48ff802a010

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (KutluOzel-b) [reply packages/zowe-explorer/src/dataset/actions.ts:1528]: @t1m0thyj I got your concern. It was clear. My question was ; should we start to implement such messages ? If so I think then for consistency we need to add other ones as well. For example do we have it this one on rename function right now ?

**Review by traeok — APPROVED** [review]
LGTM, tested locally and it is still functioning as intended. 

Just a couple quick suggestions: 
- The `showInputBox` calls can be updated to use the new `Gui` module.
- We might want to pass a `title` value (same as `pasteMember` placeholder) to these input boxes, that will clarify the purpose of those dialogs to the end-user, e.g. "Enter new name for dataset {...}"

**Review by t1m0thyj — COMMENTED** [review]
* When copying data sets and I am prompted for a new DSN, the input is rejected as invalid if the name is identical to the original. But when copying members, it accepts the original member name and doesn't require me to change it. Is this an inconsistency we should fix?
* When I try to copy data sets using an FTP profile, I get the following errors:
image
  Should it show the following error instead (that occurs when I try to copy members):
  image

**Review by traeok — CHANGES_REQUESTED** [review]
Just saw @t1m0thyj 's comments regarding the FTP (thanks for testing that case), will approve once comments are addressed.

**Comment by KutluOzel-b** [comment]
> LGTM, tested locally and it is still functioning as intended.
> 
> Just a couple quick suggestions:
> 
> * The `showInputBox` calls can be updated to use the new `Gui` module.
> * We might want to pass a `title` value (same as `pasteMember` placeholder) to these input boxes, that will clarify the purpose of those dialogs to the end-user, e.g. "Enter new name for dataset {...}"

Hey @traeok  thanks for the feedbacks. Adressed in : https://github.com/zowe/vscode-extension-for-zowe/pull/2051/commits/ea4cba23032e30d79ef7cfc8a9dba7f15687b29c

**Comment by KutluOzel-b** [comment]
Hey @t1m0thyj  Thanks for the feedbacks. 

> * When copying data sets and I am prompted for a new DSN, the input is rejected as invalid if the name is identical to the original. But when copying members, it accepts the original member name and doesn't require me to change it. Is this an inconsistency we should fix?

**If you select multiple members to copy then we are not asking for every member( I think this is the correct behaviour otherwise it would be painful to copy multiple  ) name but if single is copied then user can change it. We could have implemented such feature that checks if member is already available for the current PDS , but this is costly since we are already doing that in api**

> * When I try to copy data sets using an FTP profile, I get the following errors:
>   image
>   Should it show the following error instead (that occurs when I try to copy members):
>   image

**Since we don't have an api to do this operation, new PDS is created with the process : download members, allocate new Pds , create new local files , upload to mainframe , pull again from mf. Each of the process which is not supported by API would reject due to various reasons .I think It doesn't seems good to ignore part of a process and inform users with "can't copy" since user might find solution of this wrt authorization and such .**

**Comment by traeok** [comment]
@KutluOzel-b Thanks for addressing those comments 🙂 

I did find an API endpoint in `@zowe/zos-files-for-zowe-sdk`, and was wondering what your thoughts would be on using this instead of downloading and uploading? [Here](https://github.com/zowe/zowe-cli/blob/master/packages/zosfiles/src/methods/copy/Copy.ts#L40) is the method that I am referencing.

It seems like we should be able to do a copy operation on the mainframe, then pull the resultant dataset from that operation. Should help speed up runtime and would avoid having to download and re-upload files.

https://www.ibm.com/docs/en/zos/2.1.0?topic=interface-zos-data-set-member-utilities

**Comment by KutluOzel-b** [comment]
> @KutluOzel-b Thanks for addressing those suggestions 🙂
> 
> I did find an API endpoint in `@zowe/zos-files-for-zowe-sdk`, and was wondering what your thoughts would be on using this instead of downloading and uploading? [Here](https://github.com/zowe/zowe-cli/blob/master/packages/zosfiles/src/methods/copy/Copy.ts#L40) is the method that I am referencing.
> 
> It seems like we should be able to do a copy operation on the mainframe, then pull the resultant dataset from that operation. Should help decrease runtime for that operation and would avoid having to make local copies first. I've added a link to the API below for context:
> 
> https://www.ibm.com/docs/en/zos/2.1.0?topic=interface-zos-data-set-member-utilities
> 
> I've also proposed a similar optimization for USS, here is that [PR](https://github.com/zowe/vscode-extension-for-zowe/pull/2082) for reference.

:exploding_head::exploding_head: I wish I knew this before. I thought we could only use what is already in cli/imperative/ftp api. 

If such api is available, then we should use it . At this point we might take this as an improvement or cancel this pr .

what do you think ?

**Comment by traeok** [comment]
> 🤯🤯 I wish I knew this before. I thought we could only use what is already in cli/imperative/ftp api.
> 
> If such api is available, then we should use it . At this point we might take this as an improvement or cancel this pr .
> 
> what do you think ?

We might be able to update this PR to use that API instead, since there is a function within Zowe's z/OS files SDK for it. Either way, we can always implement that API into this feature as a future improvement, as the feature still functions as intended for this PR. Whatever works best for you 🙂

**Review by traeok — APPROVED** [review]
LGTM, thanks @KutluOzel-b 😀

File: packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts
Comment (zFernand0) [inline packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts:344]: Don't we need to make this optional to avoid breaking extenders?

**Review by zFernand0 — CHANGES_REQUESTED** [review]
left one comment about the ze-api new method

**Review by traeok — CHANGES_REQUESTED** [review]
Sorry for the confusion, I've updated my review with a note about the APIs. I edited my original comment but I probably should have posted it as a follow-up comment instead.

@zFernand0 mentioned that we will have issues using that Copy API between LPARs. We can adapt this in the future to use the new endpoint, but until we add a check for pasting between sessions it will probably be best to use the original implementation.

File: packages/zowe-explorer-ftp-extension/CHANGELOG.md
Comment (KutluOzel-b) [reply packages/zowe-explorer-ftp-extension/CHANGELOG.md:7]: I had to modify changelog to get it passed. Didn't want to add something like "added unsupported warning of copy datasets"

File: packages/zowe-explorer-api/src/profiles/ZoweExplorerZosmfApi.ts
Comment (KutluOzel-b) [reply packages/zowe-explorer-api/src/profiles/ZoweExplorerZosmfApi.ts:284]: addressed in : https://github.com/zowe/vscode-extension-for-zowe/pull/2051/commits/b62c2e4eed34167aa8faa83bb2baee2228ebc82d

File: packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts
Comment (KutluOzel-b) [reply packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts]: Good catch ! adressed in : https://github.com/zowe/vscode-extension-for-zowe/pull/2051/commits/b62c2e4eed34167aa8faa83bb2baee2228ebc82d

**Comment by KutluOzel-b** [comment]
> Sorry for the confusion, I've updated my review with a note about the APIs. I edited my original comment about this copy API, but I probably should have posted it as a follow-up comment instead.
> 
> @zFernand0 mentioned that we will have issues using that Copy API between LPARs. We can adapt this in the future to use the new endpoint, but until we add a check for pasting between sessions it will probably be best to use your original implementation. I noticed that `copyDataSet` will use this API regardless of the LPAR, so these changes might break copy/pasting between LPARs until cross-LPAR functionality is implemented for that API endpoint.

In current Implementation user doesn't asked to be choose LPAR while copying PDS and Sequential datasets and won't be a problem, but we can modify this in future once api is adjusted accordingly. 
Copying members of a PDS though, can be pasted into other LPAR and this gives the result " PDS is not available " from api.

File: packages/zowe-explorer/src/globals.ts
Comment (zFernand0) [inline packages/zowe-explorer/src/globals.ts:36]: The fact that we have gone down in the number of commands scares me a little bit.

I know the FTP extension was updated as part of this PR, but can we teat this with another extender's extension that has not updated their implementation yet?

@rudyflores or @JillieBeanSim, can you help us here?

**Review by zFernand0 — COMMENTED** [review]
Thanks for making the new API optional in the ZE-API interface :yum:
I'm still curious if other extenders may encounter small issues after this PR because of the removal of some commands :thinking:

File: packages/zowe-explorer/src/extension.ts
Comment (zFernand0) [inline packages/zowe-explorer/src/extension.ts:436]: Since we are in an LTS release, removing commands might be considered a breaking change.
We should implement this new behavior with new commands and deprecate the old ones.

Already in the middle of this 😋

**Review by zFernand0 — CHANGES_REQUESTED** [review]
This could lead to a breaking change in the middle of an LTS release.
I'll suggest some changes in a commit later today

File: packages/zowe-explorer/src/extension.ts
Comment (zFernand0) [reply packages/zowe-explorer/src/extension.ts:436]: 🤔 Removing the commands may not be a breaking change since they were never expose to the end user.
However, changing an exported function's signature is.
I'm happy to work on this in order to deprecate the copyDataSet method 😋

**Comment by zFernand0** [comment]
Known issue:
While this PR successfully created the destination sequential dataset with the same attributes as the source sequential dataset...
The underlying z/OSMF API may prevent the operation due to the record format being used.
Tested with ZE and CLI and they both return a similar error regarding `recfm = U` 
image

**Review by zFernand0 — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]

**Comment by zFernand0** [comment]
We believe that removing hidden (i.e. never exposed in the command palette) commands from the `context.subscriptions` is not a breaking change, thus we kept the removal of `copyDataSet`, `copyMember`, and `pasteMember`.

However, the change in behavior of an exported function is something we often try to avoid. For this reason, we decided the bring back the `copyDataSet` method. I took the liberty to rename the newly added methods (and their registered commands) to their plural forms (e.g. copyDataSets, pasteDataSets) since there the concept of method overloading in Typescript requires us to keep the same number of parameters (and we went from 1 to 3).

I'm happy to continue discussing these decision in a parking lot on our next call, or on our slack channel if needed 😋

**Review by zFernand0 — COMMENTED** [review]
Not sure I should be approving this PR anymore
but, LGTM! 😋

**Review by traeok — DISMISSED** [review]
LGTM! Thanks @KutluOzel-b and @zFernand0

File: packages/zowe-explorer-ftp-extension/__tests__/__unit__/Mvs/ZoweExplorerFtpMvsApi.unit.test.ts
Comment (traeok) [inline packages/zowe-explorer-ftp-extension/__tests__/__unit__/Mvs/ZoweExplorerFtpMvsApi.unit.test.ts:202]: lines 204-205 can be placed in the catch block, but the result is the same in this situation either way

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Found a few showstoppers: 
- copy is not triggered for extenders
- there is no way to replace an existing dataset or member (even though the APIs support it)

**Comment by zFernand0** [comment]
Test scenarios to try:
- [ ] Copy a PS dataset to a new PS dataset
- [ ] Copy a PS dataset to an existing dataset (Prompt: `replace dataset`)
- [ ] * Copy a PO dataset to a new PO dataset
- [ ] * Copy a PO dataset to an existing PO dataset (Prompt: `merge and replace`)
- [ ] Copy a member to a new member
- [ ] (NEW) Copy a member to an existing member (Prompt: `replace member`)

---
Scenarios not supported by ZE
- [ ] Copy a PS dataset to a new member in an existing PO dataset
- [ ] Copy a PS dataset to an existing member in an existing PO dataset
- [ ] Copy a PS dataset to a new member in a new PO dataset
- [ ] Copy a member to a new PS dataset
- [ ] Copy a member to an existing PS dataset

---
Notes:
- `*` not supported by the FTP plugin (https://github.com/zowe/zowe-cli-ftp-plugin/pull/131)
- All other scenarios are supported by the FTP plugin (https://github.com/zowe/zowe-cli-ftp-plugin/pull/131)

File: packages/zowe-explorer/src/shared/context.ts
Comment (traeok) [inline packages/zowe-explorer/src/shared/context.ts:389]: Looks like the linter wants braces for these one-line `if` statements 😅 (same w/ line 391)

File: packages/zowe-explorer/src/shared/context.ts
Comment (traeok) [inline packages/zowe-explorer/src/shared/context.ts:392]: This *might* be optimized by JavaScript at some point, but I see there are two calls to `getParent` here. If this will increase code complexity in SonarCloud by declaring the parent, then this is good as-is 👍

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (traeok) [inline packages/zowe-explorer/src/dataset/actions.ts:1662]: Should we define the type signature for the function parameter `action`? Just in case someone tries to call this function elsewhere (as it is exported)

**Review by traeok — APPROVED** [review]
LGTM! Just left a couple comments regarding lint warnings, etc

**Review by JillieBeanSim — DISMISSED** [review]
this LGTM! thanks @zFernand0 & @KutluOzel-b 

We will definitely need to document the limitations as known issues or something

File: packages/zowe-explorer/src/shared/context.ts
Comment (zFernand0) [reply packages/zowe-explorer/src/shared/context.ts:396]: I kind of forgot about this one 😢 
I don't mind storing that in a variable even though it's just a getter for the `mParent` property
```typescript
public getParent(): IZoweTreeNode {
   return this.mParent;
}
```

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:1156]: Should these strings be localized?

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:1623]: I'm not familiar with DA (Direct Access?) data sets. Has this PR been tested to verify we support them?

**Review by t1m0thyj — APPROVED** [review]

**Comment by JillieBeanSim** [comment]
> Seems like the Theia tests broke after merging from main. 😢 Looking into it 👀

It could be from the Theia version update to 1.35.0

**Comment by zFernand0** [comment]
> > Seems like the Theia tests broke after merging from main. 😢 Looking into it 👀
> 
> It could be from the Theia version update to 1.35.0

Seems like that's the case since replaying a successful run from `main`  now fails
https://github.com/zowe/vscode-extension-for-zowe/actions/runs/4235493965/jobs/7420836048

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (zFernand0) [reply packages/zowe-explorer/src/dataset/actions.ts:1625]: I was trying to understand a bit more what Direct Access (DA) datasets were and, similar to VSAM, these Basic Direct Access Method (BDAM) datasets are a special type that we don't support for copy/paste operations.

I'll remove the DA specific considerations from this PR 😋

**Review by traeok — DISMISSED** [review]
LGTM!

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/dataset/actions.ts:1572]: If copyDataSet hasn't been picked up by an extender yet we should have a check for this otherwise user will see messages like below. 
We shouldn't expect extenders to add 'Not Supported' like I see in the FTP package here, it should be handled by the ZE extension like we have in the past where if the particular API didn't exist we either 
1. handle it the old way if this is a new way
2. return the error of this action not being supported if the new api doesn't exist

Screen Shot 2023-02-27 at 11 40 19 AM

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]

File: packages/zowe-explorer/i18n/sample/src/dataset/actions.i18n.json
Comment (traeok) [inline packages/zowe-explorer/i18n/sample/src/dataset/actions.i18n.json:94]: Do we want to make this more specific to the "copy dataset" action? Something like "Copying datasets is not supported" - not a request to change, just wanted to get your thoughts on it

**Review by traeok — DISMISSED** [review]
LGTM, check was added for the API function 👍 thanks @zFernand0

**Review by traeok — APPROVED** [review]

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (zFernand0) [reply packages/zowe-explorer/src/dataset/actions.ts]: And error message will appear indicating that `Copying datasets is not supported`
However, that leaves the question of whether we want to prevent the `allocateLike` from happening or not.

As it stands, I believe the new Copy/Paste experience is a little better than before.

Do we want to address the AllocateLike (and possibly other misbehaviors that we haven't found yet) in this PR?

(Personal opinion) 
I have considered going with a similar approach as in #2082 where we rely on the clipboard instead of performing a complex, multi-step, error-prone, user interaction for such functionality. 
However, some design decisions need to be discussed. For example, the USS file system may not require allocation of resources.

**Review by JillieBeanSim — APPROVED** [review]
thanks for the updates @zFernand0 LGTM

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (JillieBeanSim) [reply packages/zowe-explorer/src/dataset/actions.ts]: I think Allocate like can be a future PR

**Pull request merged**
